### DISCUSSION TITLE
[BETA] Version 0.3.1.0021 released!

### DISCUSSION THREAD

**Discussion by zhenrong-wang** [zhenrong-wang/hpc-now#28] [Announcements]
**IMPORTANT: This upgrade is not compatible with 0.2.x**

----------------------------------

**Main Changes:**

**1. Now you can set a password for hpc-now with the installer**

Now, you can use the installer to set a password for encryption/decryption of hpc-now.  Take macOS for an example, you can run the command:

`sudo ./installer-dwn.exe setpass --pass COMPLEX_PASSWORD` to rotate the password.

**2. Strongly enhanced the robustness by rewriting general functions**

We rewrote several key general functions for the whole hpc-now service. Potential stack overflows are avoided and fixed. 

**3. Fixed several known bugs**

Including a major bug for baidubce cloud, a minor bug for NFS volume scaling up, as well as several usability bugs.

-----------------------------------------------------
*We also added a CONTRIBUTING.md to this project, looking forward to your feedback and contributions!*

*HPC-NOW: Start your HPC journey in the cloud now.*

<hr /><em>This discussion was created from the release <a href='https://github.com/zhenrong-wang/hpc-now/releases/tag/0.3.1.beta21'>[BETA] Version 0.3.1.0021 released!</a>.</em>

**Comment by chengzhejun** [comment]
How to contribute?

**Reply by zhenrong-wang** [reply]
Please check the readme and contributing guide first.
